Code/DataStructs/DiscreteValueVect.cpp

```diff
@@ -277,11 +267,7 @@ DiscreteValueVect &DiscreteValueVect::operator-=(
   for (unsigned int i = 0; i < d_length; i++) {
     unsigned int v1 = getVal(i);
     unsigned int v2 = other.getVal(i);
-    if (v1 > v2) {
-      setVal(i, v1 - v2);
-    } else {
-      setVal(i, 0);
-    }
+    setVal(i, std::max(0u, v1 - v2));
```

Comment:
This won't work. Both v1 and v2 are unsigned, so when v1 - v2 is also unsigned. That means that if v2 is > v1 tthe result of `v1-v2` is a very large number.
If you want to simplify this into a one liner (and that's not a bad thing), you could do:
```suggestion
    setVal(i, v1 > v2 ? (v1 - v2) : 0);
```